File: FoundationFramework/DictionaryExample/DictionaryExample/Profiler.m
//
//  Profiler.m
//  ArrayExample
//
//  Created by Nikolai Baklanov on 08/08/2017.
//  Copyright © 2017 Nikolai. All rights reserved.
//

#import "Profiler.h"

@implementation Profiler

+ (NSTimeInterval)performBlockWithTime:(EmptyBlock)executingBlock
{
    NSDate *startDate = [NSDate new];
    executingBlock();
    NSDate *endDate = [NSDate new];
    NSTimeInterval interval = [endDate timeIntervalSinceDate:startDate];
    
    return interval;
}

@end

File: FoundationFramework/Collections/SwiftArrayExample/SwiftArrayExample/Utils/Profiler.swift
//
//  Profiler.swift
//  SwiftArrayExample
//
//  Created by Nikolai Baklanov on 07/10/2017.
//  Copyright © 2017 Nikolai. All rights reserved.
//

import UIKit

typealias EmptyBlock = ()->Void

class Profiler: NSObject {

    static func performBlockWithTime(block:EmptyBlock) -> TimeInterval
    {
        let startDate:Date = Date.init();
        block();
        let endDate:Date = Date.init();
        return endDate.timeIntervalSince(startDate);
    }
}
